nifty/pipeline/nifsUtils.py

```diff
@@ -1460,5 +1462,151 @@ def hasBPMExt(self, file):
+                if "-ARC" not in hdul['PRIMARY'].header['DATALAB']:
+                    hdul['PRIMARY'].header['DATALAB'] = hdul['PRIMARY'].header['DATALAB']+"-ARC"
+                if ".fits" not in hdul['PRIMARY'].header['FLATIMAG']:
+                    hdul['PRIMARY'].header['FLATIMAG'] = hdul['PRIMARY'].header['FLATIMAG']+".fits"
+                hdul['PRIMARY'].header['SHFILE'] = (os.path.split(cals.shift_file)[1], self.keywordDict['SHFILE'])
+                hdul['PRIMARY'].header['ACOMBINE'] = (len(cals.arcs), self.keywordDict['ACOMBINE'])
+                for i in range(len(cals.arcs)):
+                    hdul['PRIMARY'].header['ACMB'+str(i+1)] = (os.path.split(cals.arcs[i])[1], self.keywordDict['ACMB'])
+                hdul['PRIMARY'].header['DCOMBINE'] = (len(cals.arcdarks), self.keywordDict['DCOMBINE'])
+                for i in range(len(cals.arcdarks)):
+                    hdul['PRIMARY'].header['DCMB'+str(i+1)] = (os.path.split(cals.arcdarks[i])[1], self.keywordDict['DCMB'])
+                hdul.flush()
+            if not self.hasBPMExt(cals.arc_file):
+                iraf.imcopy(cals.bpm_file, cals.arc_file+"[BPM,type=mask,append]")
+        except Exception:
+            logging.error("Problem tagging {}.".format(cals.arc_file))
+
+    def tagRonchi(self, cals):
+        try:
+            with fits.open(cals.ronchi_file, mode="update") as hdul:
+                if "-RONCHI" not in hdul['PRIMARY'].header['DATALAB']:
+                    hdul['PRIMARY'].header['DATALAB'] = hdul['PRIMARY'].header['DATALAB']+"-RONCHI"
+                if ".fits" not in hdul['PRIMARY'].header['FLATIMAG']:
+                    hdul['PRIMARY'].header['FLATIMAG'] = hdul['PRIMARY'].header['FLATIMAG']+".fits"
+                if ".fits" not in hdul['PRIMARY'].header['DARKIMAG']:
+                    hdul['PRIMARY'].header['DARKIMAG'] = hdul['PRIMARY'].header['DARKIMAG']+".fits"
+                hdul['PRIMARY'].header['SHFILE'] = (os.path.split(cals.shift_file)[1], self.keywordDict['SHFILE'])
+                hdul['PRIMARY'].header['FCOMBINE'] = (len(cals.ronchis), self.keywordDict['FCOMBINE'])
+                for i in range(len(cals.ronchis)):
+                    hdul['PRIMARY'].header['FCMB'+str(i+1)] = (os.path.split(cals.ronchis[i])[1], self.keywordDict['FCMB'])
+                hdul['PRIMARY'].header['DCOMBINE'] = (len(cals.flatdarks), self.keywordDict['DCOMBINE'])
+                for i in range(len(cals.flatdarks)):
+                    hdul['PRIMARY'].header['DCMB'+str(i+1)] = (os.path.split(cals.flatdarks[i])[1], self.keywordDict['DCMB'])
+                hdul.flush()
+            if not self.hasBPMExt(cals.ronchi_file):
+                iraf.imcopy(cals.bpm_file, cals.ronchi_file+"[BPM,type=mask,append]")
+        except Exception:
+            logging.error("Problem tagging {}.".format(cals.ronchi_file))
+
+
+    def tagShift(self, cals):
+        pass
+
+    def hasBPMExt(self, file):
+        hdulist = astropy.io.fits.open(file)
```

Comment:
Same comment about the additional astropy.io.fits.open call as in the previous commit.